geoengine/workflow.py

```diff
@@ -552,22 +560,28 @@ async def read_new_bytes() -> Optional[bytes]:
                 yield tile
 
     async def raster_stream_into_xarray(
-            self,
-            query_rectangle: QueryRectangle,
-            clip_to_query_rectangle: bool = False,
-            open_timeout: int = 60) -> xr.DataArray:
+        self,
+        query_rectangle: QueryRectangle,
+        clip_to_query_rectangle: bool = False,
+        open_timeout: int = 60,
+        bands: Optional[List[int]] = None  # TODO: move into query rectangle?
+    ) -> xr.DataArray:
         '''
         Stream the workflow result into memory and output a single xarray.
 
         NOTE: You can run out of memory if the query rectangle is too large.
         '''
 
+        if bands is None:
+            bands = [0]
```

Comment:
cf. above as well

but it also should be covered since you call `raster_stream`

Reply:
cf my other comment.

Reply:
you responded to another issue

Reply:
886fac352fcd3fe58913a8d40fb193e617e67d7d

Reply:
Ok, but is it necessary to catch this twice? It is covered by raster_stream, right?

Reply:
true, but can't hurt to check it again, as it might also be used for something else than callung raster_stream in the future?